pkg/cli/server/server.go

```diff
@@ -280,14 +300,15 @@ func run(app *cli.Context, cfg *cmds.Server) error {
 	agentConfig.DataDir = filepath.Dir(serverConfig.ControlConfig.DataDir)
 	agentConfig.ServerURL = url
 	agentConfig.Token = token
-	agentConfig.DisableLoadBalancer = true
+	agentConfig.DisableLoadBalancer = !serverConfig.ControlConfig.DisableAPIServer
+	agentConfig.ETCDAgent = serverConfig.ControlConfig.DisableAPIServer
+
 	agentConfig.Rootless = cfg.Rootless
-	if agentConfig.Rootless {
```

Comment:
Do we not still need to guard against unsharing twice?

Reply:
fixed